diff --git a/addon.xml b/addon.xml
index 943e68b..ec36521 100644
--- a/addon.xml
+++ b/addon.xml
@@ -2,8 +2,14 @@
 <addon id="plugin.video.1channel" version="2.0.1" name="1Channel" provider-name="Bstrdsmkr">
     <requires>
         <import addon="xbmc.python" version="2.0"/>
+        <import addon="script.module.xbmcaddon"    version="1.1.0"/>
+        <import addon="script.module.urlresolver"  version="0.0.1"/>
+        <import addon="script.module.simplejson"   version="2.0.0"/>
         <import addon="script.module.urlresolver"  version="1.0.0"/>
         <import addon="script.module.t0mm0.common" version="1.1.0"/>
+        <import addon="script.module.pysqlite2"    version="1.0.0"/>
+        <import addon="script.module.elementtree"  version="1.2.0"/>
+        <import addon="script.module.xbmcvfs"      version="1.0.1"/>
 		<import addon="script.module.metahandler"  version="1.0.0"/>
 		<import addon="script.module.myconnpy"     version="0.3.2"/>
     </requires>
@@ -17,4 +23,9 @@
         <description>This addon is supported at [COLOR blue]www.XBMCHub.com[/COLOR],  [COLOR white][B]NOT[/B][/COLOR] [COLOR red]xbmc.org[/COLOR].  Play a wide selection of movies and TV shows from 1channel.ch</description>
         <platform>all</platform>
     </extension>
+    <statfeed>
+        <working>mostly</working>
+        <notes>Images for hosts requiring captcha dont appear</notes>
+        <maintainer>kozz</maintainer>
+    </statfeed>
 </addon>
\ No newline at end of file
diff --git a/default.py b/default.py
index 9953366..3555ab4 100644
--- a/default.py
+++ b/default.py
@@ -33,7 +33,7 @@ from t0mm0.common.addon import Addon
 from t0mm0.common.net import Net
 from metahandler import metahandlers
 from metahandler import metacontainers
-from operator import itemgetter, methodcaller
+from operator import itemgetter
 
 import metapacks
 from utils import *
@@ -88,7 +88,7 @@ prepare_zip = False
 metaget=metahandlers.MetaData(preparezip=prepare_zip)
 
 if not xbmcvfs.exists(addon.get_profile()):
-     xbmcvfs.mkdirs(addon.get_profile())
+     xbmcvfs.mkdir(addon.get_profile())
 
 
 def art(file):
@@ -231,7 +231,7 @@ def GetURL(url, params=None, referrer=BASE_URL, silent=False, cache_limit=8):
     if referrer: req.add_header('Referer', referrer)
     
     try:
-        response = urllib2.urlopen(req, timeout=10)
+        response = urllib2.urlopen(req)
         body = response.read()
         body = unicode(body,'iso-8859-1')
         h = HTMLParser.HTMLParser()
@@ -989,7 +989,7 @@ def migrate_favs_to_web():
     # failures = [('title1','url1'), ('title2','url2'), ('title3','url3'), ('title4','url4'), ('title5','url5'), ('title6','url6'), ('title7','url7')]
     if ret:
         if failures:
-            params = ', '.join('%s' if DB=='mysql' else '?' for item in failures)
+            params = ', '.join('?' for item in failures)
             sql_delete = 'DELETE FROM favorites WHERE url NOT IN (SELECT url FROM favorites WHERE url IN (%s))'
             sql_delete = sql_delete % params
             addon.log(sql_delete)
@@ -1103,13 +1103,14 @@ def zipdir(basedir, archivename):
     from contextlib import closing
     from zipfile import ZipFile, ZIP_DEFLATED
     assert os.path.isdir(basedir)
-    with closing(ZipFile(archivename, "w", ZIP_DEFLATED)) as z:
-        for root, dirs, files in os.walk(basedir):
-            #NOTE: ignore empty directories
-            for fn in files:
-                absfn = os.path.join(root, fn)
-                zfn = absfn[len(basedir)+len(os.sep):] #XXX: relative path
-                z.write(absfn, zfn)
+    z = ZipFile(archivename, "w", ZIP_DEFLATED)
+    for root, dirs, files in os.walk(basedir):
+        #NOTE: ignore empty directories
+        for fn in files:
+            absfn = os.path.join(root, fn)
+            zfn = absfn[len(basedir)+len(os.sep):] #XXX: relative path
+            z.write(absfn, zfn)
+    z.close
 
 def extract_zip(src, dest):
         try:
@@ -1632,10 +1633,17 @@ def multikeysort(items, columns, functions={}, getter=itemgetter):
     """
     comparers = []
     for col in columns:
-        column = col[1:] if col.startswith('-') else col
+        if col.startswith('-'):
+            column = col[1:]
+        else:
+            colume = col
         if not column in functions:
             functions[column] = getter(column)
-        comparers.append((functions[column], 1 if column == col else -1))
+        if column == col:
+            comparers.append((functions[column], 1))
+        else:
+            comparers.append((functions[column], -1))
+
 
     def comparer(left, right):
         for func, polarity in comparers:
@@ -1879,6 +1887,7 @@ def build_listitem(video_type, title, year, img, resurl, imdbnum='', season='',
 
 
 def upgrade_db():
+    return
     addon.log('Upgrading db...')
     for table in ('subscriptions', 'favorites'):
         sql = "UPDATE %s SET url = replace(url, 'http://www.1channel.ch', '')" %table
@@ -1889,7 +1898,6 @@ def upgrade_db():
         db.commit()
         db.close()
 
-
 def fix_existing_strms():
     for folder in ('tvshow-folder', 'movie-folder'):
         save_path = addon.get_setting(folder)
@@ -1899,17 +1907,19 @@ def fix_existing_strms():
                 if currentFile.endswith('.strm'):
                     addon.log( "processing file: %s" %currentFile)
                     full_path = os.path.join(root, currentFile)
-                    with open(full_path, 'r+') as target:
-                        content = target.read()
-                        target.seek(0)
-                        new_content = content.replace('&url=http%3A%2F%2Fwww.1channel.ch', '&url=')
-                        if not '&video_type=' in new_content:
-                            if folder == 'tvshow-folder':
-                                new_content += '&video_type=episode'
-                            elif folder == 'movie-folder':
-                                new_content += '&video_type=movie'
-                        addon.log('Writing new content: %s' %new_content)
-                        target.write(new_content)
+                    target = open(full_path, 'r+')
+                    content = target.read()
+                    target.seek(0)
+                    new_content = content.replace('&url=http%3A%2F%2Fwww.1channel.ch', '&url=')
+                    if not '&video_type=' in new_content:
+                        if folder == 'tvshow-folder':
+                            new_content += '&video_type=episode'
+                        elif folder == 'movie-folder':
+                            new_content += '&video_type=movie'
+                    addon.log('Writing new content: %s' %new_content)
+                    target.write(new_content)
+                    target.close
+
                     
 
 mode        = addon.queries.get('mode',       None)
@@ -2024,4 +2034,4 @@ elif mode=='migrateFavs':
     migrate_favs_to_web()
 elif mode=='Help':
     addon.log('Showing help popup')
-    popup = TextBox()
\ No newline at end of file
+    popup = TextBox()
diff --git a/resources/settings.xml b/resources/settings.xml
index 2952e06..b17f9b6 100644
--- a/resources/settings.xml
+++ b/resources/settings.xml
@@ -1,6 +1,6 @@
 <settings>
    <category label="Meta Data">
-  	<setting id="use-meta" type="bool" label="30600" default="true"/>
+  	<setting id="use-meta" type="bool" label="30600" default="false"/>
 	<setting id="meta-path" type="action" label="30601" enable="!eq(-1,false)" action="RunPlugin(plugin://plugin.video.1channel/default.py?mode=9988)" option="close"/>
 	<setting id="enable-fanart" type="bool" label="30602" enable="!eq(-2,false)" default="true"/>
 	<setting id="use-posters" type="bool" label="30603" enable="!eq(-3,false)" default="false"/>
